### DISCUSSION TITLE
Differentiate c and c++ in configuration?

### DISCUSSION THREAD

**Discussion by jalanh11640** [microsoft/vscode-cpptools#10430] [Q&A]
Our makefile system differentiates between compilation of c and c++ files and provides different include paths, defines and options to the compiler.

Is there a way to account for this in a single configuration in c_cpp_properties.json or is the only option to create two configurations (say config-c and config-cpp) and switch between them whether editing c or c++ files?  That seems less than ideal.

What's the solution here?

Thanks,
Jeff

**Comment by sean-mcmanus** [comment]
Having tried installing the Makefile Tools extension and setting configurationProvider to "ms-vscode.makefile-tools"? It's theoretically possible they could send the correct configuration based on the file extension type. If not, you could file a bug on that extension.
